### PR TITLE
RPP Updates

### PR DESCRIPTION
**Pull request by kiritigowda**
* CentOS 7 Build
* CMakeList updates

### CODE REVIEW

**Comment by kiritigowda** [comment]
@LakshmiKumar23 need this to be merged for the CI to pass - https://github.com/ROCmSoftwarePlatform/rocJenkins/pull/445

**Comment by LakshmiKumar23** [comment]
@kiritigowda Trying to build a centos docker to test this. Using (this docker)[https://github.com/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/blob/master/docker/centos7/level-4.dockerfile]. Unable to build

```
 You could try using --skip-broken to work around the problem
Error: Package: hsakmt-roct-devel-20220426.0.86.50200-65.el7.x86_64 (ROCm)
           Requires: libdrm-amdgpu
 You could try running: rpm -Va --nofiles --nodigest
The command '/usr/bin/scl enable devtoolset-7 echo -e "[ROCm]\nname=ROCm\nbaseurl=https://repo.radeon.com/rocm/yum/rpm\nenabled=1\ngpgcheck=1\ngpgkey=https://repo.radeon.com/rocm/rocm.gpg.key" >         /etc/yum.repos.d/rocm.repo && yum -y install --nogpgcheck rocm-dev' returned a non-zero code: 1
```

**Comment by kiritigowda** [comment]
@LakshmiKumar23 use this docker - compute-artifactory.amd.com:5000/rocm-plus-docker/compute-rocm-rel-5.2:52-centos-7.7-stg1 and follow the setup instructions

**Comment by LakshmiKumar23** [comment]
> @LakshmiKumar23 use this docker - `compute-artifactory.amd.com:5000/rocm-plus-docker/compute-rocm-rel-5.2:52-centos-7.7-stg1` and follow the setup instructions

@kiritigowda Followed the steps on this docker. Issue with clang
```
cmake -DBACKEND=HIP ../
-- The CXX compiler identification is unknown
CMake Error at CMakeLists.txt:27 (project):
  The CMAKE_CXX_COMPILER:

    clang++

  is not a full path and was not found in the PATH.

  Tell CMake where to find the compiler by setting either the environment
  variable "CXX" or the CMake cache entry CMAKE_CXX_COMPILER to the full path
  to the compiler, or to the compiler name if it is in the PATH.


-- Configuring incomplete, errors occurred!
See also "/var/tmp/boost_1_72_0/rpp/build/CMakeFiles/CMakeOutput.log".
See also "/var/tmp/boost_1_72_0/rpp/build/CMakeFiles/CMakeError.log".
```

**Comment by kiritigowda** [comment]
> > @LakshmiKumar23 use this docker - `compute-artifactory.amd.com:5000/rocm-plus-docker/compute-rocm-rel-5.2:52-centos-7.7-stg1` and follow the setup instructions
> 
> @kiritigowda Followed the steps on this docker. Issue with clang
> 
> ```
> cmake -DBACKEND=HIP ../
> -- The CXX compiler identification is unknown
> CMake Error at CMakeLists.txt:27 (project):
>   The CMAKE_CXX_COMPILER:
> 
>     clang++
> 
>   is not a full path and was not found in the PATH.
> 
>   Tell CMake where to find the compiler by setting either the environment
>   variable "CXX" or the CMake cache entry CMAKE_CXX_COMPILER to the full path
>   to the compiler, or to the compiler name if it is in the PATH.
> 
> 
> -- Configuring incomplete, errors occurred!
> See also "/var/tmp/boost_1_72_0/rpp/build/CMakeFiles/CMakeOutput.log".
> See also "/var/tmp/boost_1_72_0/rpp/build/CMakeFiles/CMakeError.log".
> ```

@LakshmiKumar23 -- Need to setup clang

```
sudo yum update
sudo yum install git cmake3
yum install llvm-toolset-7-clang llvm-toolset-7-clang-analyzer llvm-toolset-7-clang-tools-extra
scl enable llvm-toolset-7 bash
```

**Comment by LakshmiKumar23** [comment]
> > > @LakshmiKumar23 use this docker - `compute-artifactory.amd.com:5000/rocm-plus-docker/compute-rocm-rel-5.2:52-centos-7.7-stg1` and follow the setup instructions
> > 
> > 
> > @kiritigowda Followed the steps on this docker. Issue with clang
> > ```
> > cmake -DBACKEND=HIP ../
> > -- The CXX compiler identification is unknown
> > CMake Error at CMakeLists.txt:27 (project):
> >   The CMAKE_CXX_COMPILER:
> > 
> >     clang++
> > 
> >   is not a full path and was not found in the PATH.
> > 
> >   Tell CMake where to find the compiler by setting either the environment
> >   variable "CXX" or the CMake cache entry CMAKE_CXX_COMPILER to the full path
> >   to the compiler, or to the compiler name if it is in the PATH.
> > 
> > 
> > -- Configuring incomplete, errors occurred!
> > See also "/var/tmp/boost_1_72_0/rpp/build/CMakeFiles/CMakeOutput.log".
> > See also "/var/tmp/boost_1_72_0/rpp/build/CMakeFiles/CMakeError.log".
> > ```
> 
> @LakshmiKumar23 -- Need to setup clang
> 
> ```
> sudo yum update
> sudo yum install git cmake3
> yum install llvm-toolset-7-clang llvm-toolset-7-clang-analyzer llvm-toolset-7-clang-tools-extra
> scl enable llvm-toolset-7 bash
> ```

@kiritigowda  Yes, I ran all the steps in the instructions

**Comment by LakshmiKumar23** [comment]
> scl enable llvm-toolset-7 bash

@kiritigowda Retried the build. It works. Must have missed a step last time. Can be merged

**Review by LakshmiKumar23 — APPROVED** [review]

**Review by asalmanp — APPROVED** [review]

File: CMakeLists.txt [hunk 1]
```diff
@@ -172,34 +176,29 @@ if("${BACKEND}" STREQUAL "HIP")
             set(HIP_PATH $ENV{HIP_PATH} CACHE PATH "Path to which HIP has been installed")
         endif()
     endif()
-    list(APPEND CMAKE_MODULE_PATH ${ROCM_PATH}/${CMAKE_INSTALL_LIBDIR}/cmake/hip ${HIP_PATH}/cmake)
+    list(APPEND CMAKE_PREFIX_PATH ${ROCM_PATH}/hip ${ROCM_PATH})
```
Comment (kiritigowda) [inline CMakeLists.txt:179]: @asalmanp changes for HIP here for review.

File: CMakeLists.txt [hunk 2]
```diff
@@ -172,34 +176,29 @@ if("${BACKEND}" STREQUAL "HIP")
...
+        # TBD: Set external flag to notify GPU_SUPPORT and HIP backend availability - Compatible with rocAL and MIVisionX
+        add_definitions(-DGPU_SUPPORT)
+        # For cmakedefine01 in rpp_backend.h.in
+        set(RPP_BACKEND_HIP 1)
+        # To use RPP_BACKEND_HIP
+        add_definitions(-DRPP_BACKEND_HIP=${RPP_BACKEND_HIP})
+        set(COMPILER_FOR_HIP ${ROCM_PATH}/llvm/bin/clang++)
+        include_directories(${ROCM_PATH}/${CMAKE_INSTALL_INCLUDEDIR})
+        link_directories(${HIP_PATH}/${CMAKE_INSTALL_LIBDIR})
+        # TBD: Link To Host/Device HIP -- hip::host/hip::device
```
Comment (kiritigowda) [inline CMakeLists.txt:192]: @asalmanp the module is built differently here. The host&device code is combined. Let me know what you think.

Comment (asalmanp) [reply CMakeLists.txt:192]: @kiritigowda you can replace the {HIP_PATH}/${CMAKE_INSTALL_LIBDIR}/libamdhip64.so with hip::host on this CMakeLists.txt (line 193) file as this is only for hip host APIs. The hip device kernels are built with this [CMakeLists.txt](https://github.com/GPUOpen-ProfessionalCompute-Libraries/rpp/blob/master/src/modules/CMakeLists.txt) which you already added the target_link_libraries(${PROJECT_NAME} hip::device) correctly.

File: CMakeLists.txt [hunk 3]
```diff
@@ -176,34 +176,29 @@ if("${BACKEND}" STREQUAL "HIP")
...
-            message("-- ${Yellow}${PROJECT_NAME} set to build with CPU backend${ColourReset}")
-        endif()
+        message("-- ${White}${PROJECT_NAME} -- Using HIP - Path:" ${HIP_PATH} "\tVersion:" ${HIP_VERSION} "\tCompiler:" ${HIP_COMPILER}${ColourReset})
+        # TBD: Set external flag to notify GPU_SUPPORT and HIP backend availability - Compatible with rocAL and MIVisionX
+        add_definitions(-DGPU_SUPPORT)
+        # For cmakedefine01 in rpp_backend.h.in
+        set(RPP_BACKEND_HIP 1)
+        # To use RPP_BACKEND_HIP
+        add_definitions(-DRPP_BACKEND_HIP=${RPP_BACKEND_HIP})
+        set(COMPILER_FOR_HIP ${ROCM_PATH}/llvm/bin/clang++)
```
Comment (kiritigowda) [inline CMakeLists.txt]: @asalmanp is PR failed to build before

**Review by asalmanp — APPROVED** [review]

**Pull request merged**